teapots/common/ndk_helper/CMakeLists.txt

```diff
@@ -1,16 +1,44 @@
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wall -fno-exceptions -fno-rtti")
+# build native_app_glue as a static lib
+cmake_minimum_required(VERSION 3.4.1)
 
-add_library(ndk-helper STATIC
-            gestureDetector.cpp
-            gl3stub.cpp
-            GLContext.cpp
-            interpolator.cpp
-            JNIHelper.cpp
-            perfMonitor.cpp
-            sensorManager.cpp
-            shader.cpp
-            tapCamera.cpp
-            vecmath.cpp)
+include(AndroidNdkModules)
+android_ndk_import_module_native_app_glue()
 
-target_include_directories(ndk-helper PRIVATE
-                           ${ANDROID_NDK}/sources/android/native_app_glue)
+add_library(NdkHelper
+  STATIC
+    gestureDetector.cpp
+    gl3stub.cpp
+    GLContext.cpp
+    interpolator.cpp
+    JNIHelper.cpp
+    perfMonitor.cpp
+    sensorManager.cpp
+    shader.cpp
+    tapCamera.cpp
+    vecmath.cpp
+)
+set_target_properties(NdkHelper
+  PROPERTIES
+    CXX_STANDARD 11
+    CXX_STANDARD_REQUIRED YES
+    CXX_EXTENSIONS NO
+    INTERFACE_INCLUDE_DIRECTORIES $<TARGET_PROPERTY:NdkHelper,INCLUDE_DIRECTORIES>
+)
+target_compile_options(NdkHelper
+  PUBLIC
+    -Wall -Werror -fno-exceptions -fno-rtti
```

Comment:
To be crystal clear: I will not be approving this change until either this goes or I hear an argument for why this is an improvement beyond "this slide deck says so".

None of these flags should be exported from this module because they are not requirements for using the module. Doing this forces these flags for all *users* of this library despite the fact that they are only relevant for *building* this library. These are policy decisions that should be made either by each library or by the builder for the project as a whole (which is what `CMAKE_CXX_FLAGS` is for).

Reply:
Yeah, I agree that this is module level and this could be done as PRIVATE.  so it limits to this module.  Client could do their own settings for the similar stuff but with different settings.

I am ok to change it to private, and have project to have the similar settings to avoid that global CMAKE_CXX_FLAGS setting, how is that?

no-exceptions and no-rtti was "have to be " used in early days (as I was told) for avoiding build issues and code size ( I think ) so I am not really ready to get rid of them here.